README.md

```diff
@@ -31,167 +31,118 @@ Where `<snapshot version>` refers to the version in the Sonatype Snapshot badge
 
 ```scala
-deleteById(IndexName("index"), DocumentId("documentId")).routing(Routing("routing")).refreshTrue
+for {
+  _ <- Elasticsearch.execute(bulk(requests: _*)) 
+} yield ()
 ```
 
-Just like `IndexName`, `Routing` is a new type that mustn't be an empty string.
+### Usage of ZIO Schema and its accessors for type-safety
 
-And if we wanted to specify lower and upper bounds for a `range` query:
+To provide type-safety in your Elasticsearch requests, ZIO Elasticsearch uses ZIO Schema. Here is an example of creating a schema for the custom type `User` and using an implicit schema to create accessors that result in type-safe requests.
 
 ```scala
-range(EmployeeDocument.age).gte(18).lt(100)
-```
+final case class Address(street: String, number: Int)
 
-### Elastic Executor
+object Address {
+  implicit val schema: Schema.CaseClass2[String, Int, Address] =
+    DeriveSchema.gen[Address]
 
-In order to get the functional effect of executing a specified Elasticsearch request, you must call the `execute` method defined on it, which returns a `ZIO` that requires an `ElasticExecutor`, fails with a `Throwable` and returns the relevant value `A` for that request.
-
-Elastic requests for creating and deleting return `CreationOutcome` and `DeletionOutcome` respectively if no other meaningful value could be returned, notifying us on the success of the request. Any other kind of error is returned as a `Throwable` in the error channel of `ZIO` for that Elastic request.
-
-If you want to execute multiple Elasticsearch requests in a single API call, you need to use the `bulk` method on those Elastic requests, and call `execute` on that bulk request instead.
-
-To provide the dependency on `ElasticExecutor`, you must pass one of the `ZLayer`s from the following import:
-
-```scala
-import zio.elasticsearch.ElasticExecutor
-```
+  val (street, number) = schema.makeAccessors(FieldAccessorBuilder)
+}
 
-For example, if you want to execute requests on a server running on `localhost` and port `9200`, you can provide the `live` ZLayer to your effect, along with a `SttpBackend` and an `ElasticConfig` layer:
+final case class User(id: Int, address: Address)
 
-```scala
-import sttp.client3.SttpBackend
-import sttp.client3.httpclient.zio.HttpClientZioBackend
-import zio.elasticsearch.{ElasticConfig, ElasticExecutor}
+object User {
+  implicit val schema: Schema.CaseClass2[String, Address, User] =
+    DeriveSchema.gen[User]
 
-val effect: RIO[ElasticExecutor, Boolean] = exists(IndexName("index"), DocumentId("document")).execute
+  val (id, address) = schema.makeAccessors(FieldAccessorBuilder)
+}
 
-effect.provide(
-  HttpClientZioBackend.layer(),
-  ZLayer.succeed(ElastichConfig("localhost", 9200)) >>> ElasticExecutor.live,
-)
+for {
+  _ <- Elasticsearch.execute(
+    search(
+      IndexName("index"),
+      must(range(User.id).gte(7).lt(10)).should(startsWith(User.address / Address.street, "ZIO"))
+    ).aggregate(aggregation)
+  )
+} yield ()
```

Comment:
Perhaps the for-comprehensions is unnecessary here as it's essentially doing a `.map(_ => ())`